### ISSUE TITLE
Entire NIP-11 dataset discarded because of one bad field

### ISSUE THREAD

**Issue by bu5hm4nn** [mikedilger/gossip#463]
```
2023-08-17T18:41:09.118934Z  WARN src/overlord/minion/mod.rs:125: wss://nostr.mom/: Unable to parse response as NIP-11 (Invalid Public Key Prefix at line 1 column 227): {"contact":"unset","description":"This is a strfry instance.","name":"strfry default","pubkey":"unset","software":"git+https://github.com/hoytech/strfry.git","supported_nips":[1,2,4,9,11,12,16,20,22,28,33,40],"version":"0.9.2"}
```

So right now there are quite a few entries in my log where there is "unset" as a pubkey in the contact fields. Currently this will result in the entire NIP-11 dataset being discarded. Since fields are Option<> type anyways, there should be no need to discard the entire NIP-11 dataset just because one field is bad.

I don't know why people do this, but I guess it is because some of the relay software (strfry in this case) requires to put something in the field?

**Comment by mikedilger** [comment]
Yeah we can make it None if the data is invalid. Seems like a good idea.

**Comment by mikedilger** [comment]
fixed in  8bca415866779fbfa1ebd875c209d6e0e7c18c3d
